src/index.ts

```diff
@@ -164,6 +194,7 @@ export class VisualObserver {
     this.#resizeObserver.unobserve(target);
 
     el.io?.disconnect();
+    el.io = null;
```

Comment:
Making sure that we clean up the intersection observer better! Although this one might not be needed.